### DISCUSSION TITLE
How to make ui.column occupy the remaining space in the vertical direction？

### DISCUSSION THREAD

**Discussion by hu76589** [zauberzeug/nicegui#1686] [Q&A] [answered]
### Question

In the Modularization Example's theme.py:

```py
with ui.column().classes('absolute-center items-center')
    yield
```

What content should I set for the class?
because I want to place an aggrid under the header to fill the remaining space on the page, and do not display vertical scrollbars on the page.

thank you!

**Comment by falkoschindler** [comment]
Sorry, @hu76589, I don't understand your question.
Can you, please, show a [minimal reproducible code example](https://en.wikipedia.org/wiki/Minimal_reproducible_example) and explain what exactly you're trying to achieve? Thanks!

**Reply by hu76589** [reply]
I just made some modifications in the example 'Modularization',
theme.py:
```py
from contextlib import contextmanager

from menu import menu

from nicegui import ui


@contextmanager
def frame(navtitle: str):
    '''Custom page frame to share the same styling and behavior across all pages'''
    ui.colors(primary='#6E93D6', secondary='#53B689', accent='#111B1E', positive='#53B689')
    with ui.header().classes('justify-between text-white'):
        ui.label('Modularization Example').classes('font-bold')
        ui.label(navtitle)
        with ui.row():
            menu()
    with ui.column().classes('w-full items-stretch'):
        yield
```

home_page.py:
```py
from nicegui import ui


def content() -> None:
    grid = ui.aggrid({
        'defaultColDef': {'flex': 1},
        'columnDefs': [
            {'headerName': 'Name', 'field': 'name'},
            {'headerName': 'Age', 'field': 'age'},
            {'headerName': 'Parent', 'field': 'parent', 'hide': True},
        ],
        'rowData': [
            {'name': 'Alice', 'age': 18, 'parent': 'David'},
            {'name': 'Bob', 'age': 21, 'parent': 'Eve'},
            {'name': 'Carol', 'age': 42, 'parent': 'Frank'},
        ],
        'rowSelection': 'multiple',
    }).classes('grow')
```

i want the aggrid to grow bigger,just like the red rectangle,fill the remaining space on the page:
![image](https://github.com/zauberzeug/nicegui/assets/92908627/af519934-3ab5-4efc-9813-224f388439a2)

**Comment by rodja** [comment]
You can remove the default padding in x-direction from the main content with 

```py
ui.query('.nicegui-content').classes('px-0')
```

**Reply by falkoschindler** [reply]
I think the question is about filling the _vertical_ space. But I'm struggling with a simple and generic solution...

**Reply by hu76589** [reply]
i try this
`ui.query('.nicegui-content').classes('absolute-full')`
and it works,but it doesn't feel too formal, I hope Nicegui can have an official solution in the future.

This is what i want
![image](https://github.com/zauberzeug/nicegui/assets/92908627/8eccb9df-b0d6-4511-97f7-0593b7013c74)

**Reply by falkoschindler** [reply]
Nice! I like this solution.
Regarding an official solution: I wonder how the API might look like. Do you have something in mind?

**Reply by hu76589** [reply]
The problem with the above solution is, '.nicegui-content' is automatically generated and may change.
we have ui.header, ui.footer, ui.left_drawer, ui.right_drawer and ui.page_sticky, but not ui.page_content, so there seems to be no suitable place to implement this function.
How about adding an ui.page_content? But except for setting the height, there are no other functions to add.

**Reply by falkoschindler** [reply, marked as answer]
The thing is that `ui.header`, `ui.footer`, `ui.left_drawer`, `ui.right_drawer` and `ui.page_sticky` all _create_ an element, while `ui.page_content` wouldn't.

But actually there is a way to access the page content without using `ui.query`:

```py
from nicegui import Client, ui

@ui.page('/')
def index_page(client: Client):
    with ui.header():
        ui.label('Header')
    client.content.classes('absolute-full')
    ui.aggrid({}).classes('h-full')
```

The only caveat is the need for the `client` object, which is available within page functions. Alternatively you can use `get_client()`:

```py
from nicegui import globals, ui

with ui.header():
    ui.label('Header')

globals.get_client().content.classes('absolute-full')
ui.aggrid({}).classes('h-full')
```

The `globals` module is more or less undocumented and subject to change. But so far it's a working solution.

**Comment by falkoschindler** [comment]
Just for reference, this is one solution @hu76589 found:

```py
with ui.header():
    ui.label('Header')

ui.query('.nicegui-content').classes('absolute-full')
ui.aggrid({}).classes('h-full')
```
